modules/gapi/include/opencv2/gapi/infer/ov.hpp

```diff
@@ -0,0 +1,357 @@
+        return *this;
+    }
+
+    /** @overload
+
+    @param layout_map Map of pairs: name of corresponding output layer
+    and its layout in OpenCV format (CV_8U, CV_32F, ...)
+    @return reference to this parameter structure.
+    */
+    Params<Net>&
+    cfgInTensorLayout(detail::ParamDesc::Map<std::string> layout_map) {
+        desc.input_tensor_layout = std::move(layout_map);
+        return *this;
+    }
+
+    /** @brief Specifies the output layout for model.
+
+    The function is used to set an output layout for model.
+
+    @param layout Precision in OpenCV format (CV_8U, CV_32F, ...)
+    will be applied to all output layers.
+    @return reference to this parameter structure.
+    */
+    Params<Net>& cfgInModelLayout(const std::string &layout) {
+        desc.input_model_layout = layout;
+        return *this;
+    }
+
+    /** @overload
+
+    @param layout_map Map of pairs: name of corresponding output layer
+    and its layout in OpenCV format (CV_8U, CV_32F, ...)
+    @return reference to this parameter structure.
+    */
+    Params<Net>&
+    cfgInModelLayout(detail::ParamDesc::Map<std::string> layout_map) {
+        desc.input_model_layout = std::move(layout_map);
+        return *this;
+    }
+
+    // BEGIN(G-API's network parametrization API)
+    GBackend      backend() const { return cv::gapi::ov::backend(); }
+    std::string   tag()     const { return Net::tag(); }
+    cv::util::any params()  const { return { desc }; }
+    // END(G-API's network parametrization API)
+
+protected:
+    detail::ParamDesc desc;
+};
+
+/*
+* @brief This structure provides functions for generic network type that
+* fill inference parameters.
+* @see struct Generic
+*/
+template<>
+class Params<cv::gapi::Generic> {
+public:
+    Params(const std::string &tag,
+           const std::string &xml,
+           const std::string &bin,
+           const std::string &device)
+        : m_tag(tag),
+          m_desc{ detail::ParamDesc::Kind{detail::ParamDesc::IR{xml, bin}}
+                , device
+                , true /* is generic */
+                , 0u
+                , 0u
+                , {} /* output_names */
+                , {} /* input_names */
+                , {} /* config */
+                , {} /* output_precision */
+                , {} /* input_tensor_layout */
+                , {} /* input_model_layout */ } {
+    }
+
+    Params(const std::string &tag,
+           const std::string &blob,
+           const std::string &device)
+        : m_tag(tag),
+          m_desc{ detail::ParamDesc::Kind{detail::ParamDesc::Blob{blob}}
+                , device
+                , true /* is generic */
+                , 0u
+                , 0u
+                , {} /* output_names */
+                , {} /* input_names */
+                , {} /* config */
+                , {} /* output_precision */
+                , {} /* input_tensor_layout */
+                , {} /* input_model_layout */ } {
```

Comment:
This is the antipattern which grew organically the last time, but is there a chance to fix it now in the new code?

Reply:
Done